<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsForm</class>
 <widget class="QWidget" name="SettingsForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>350</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="general">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QScrollArea" name="scrollArea_2">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_3">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>476</width>
            <height>152</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>10</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>10</number>
           </property>
           <item>
            <widget class="QCheckBox" name="autoHideTopBar">
             <property name="text">
              <string>Auto-hide top bar</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="preferLists">
             <property name="text">
              <string>Prefer list layouts over grid layouts</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <property name="spacing">
              <number>15</number>
             </property>
             <item>
              <widget class="QLabel" name="appStyleLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Application style</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="appStyle">
               <item>
                <property name="text">
                 <string>Default</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="darkTheme">
               <property name="text">
                <string>Dark</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <property name="spacing">
              <number>15</number>
             </property>
             <item>
              <widget class="QPushButton" name="clearCache">
               <property name="text">
                <string>Clear Cache</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="imageCaching">
               <property name="toolTip">
                <string>Not all images are cached!</string>
               </property>
               <property name="text">
                <string>Enable image caching</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_7">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="plugins">
      <attribute name="title">
       <string>Plugins</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_16">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="browsePluginsButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Browse Plugins</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="addPluginButton">
           <property name="text">
            <string>Add Plugin</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Orientation::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QListWidget" name="pluginsListWidget"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="player">
      <attribute name="title">
       <string>Player</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QScrollArea" name="scrollArea">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>476</width>
            <height>249</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>10</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>10</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_15">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_9">
               <property name="spacing">
                <number>6</number>
               </property>
               <property name="rightMargin">
                <number>5</number>
               </property>
               <item>
                <widget class="QLabel" name="externalPlayerLabel">
                 <property name="text">
                  <string>External player</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="externalPlayerEdit">
                 <property name="placeholderText">
                  <string>/usr/bin/mpv %U</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="externalPlayerButton">
                 <property name="text">
                  <string>...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_10">
               <item>
                <spacer name="horizontalSpacer_8">
                 <property name="orientation">
                  <enum>Qt::Orientation::Horizontal</enum>
                 </property>
                 <property name="sizeType">
                  <enum>QSizePolicy::Policy::Fixed</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QLabel" name="externalPlayerHint">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>%U is substituted with the video URL, %P is substituted with progress in seconds. Player must have youtube-dl/yt-dlp integration.</string>
                 </property>
                 <property name="wordWrap">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="vaapi">
             <property name="text">
              <string>Enable VA-API hardware acceleration</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="h264Only">
             <property name="toolTip">
              <string>Enforces serving of H.264 encoded content, if available. May improve playback performance and resource usage, especially on older devices. There is no guarantee this setting will be implemented by any given plugin.</string>
             </property>
             <property name="text">
              <string>Use H.264 only where possible</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_17">
             <property name="spacing">
              <number>10</number>
             </property>
             <item>
              <widget class="QLabel" name="preferredVolumeLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>There is no guarantee this setting will be implemented by any given plugin.</string>
               </property>
               <property name="text">
                <string>Preferred volume</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="preferredVolume">
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="volumeFromPlayer">
               <property name="toolTip">
                <string>There is no guarantee this setting will be implemented by any given plugin.</string>
               </property>
               <property name="text">
                <string>Set from player</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_11">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_18">
             <property name="spacing">
              <number>10</number>
             </property>
             <item>
              <widget class="QLabel" name="preferredQualityLabel">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>There is no guarantee this setting will be implemented by any given plugin.</string>
               </property>
               <property name="text">
                <string>Preferred quality</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="preferredQuality">
               <item>
                <property name="text">
                 <string>Auto</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>4320p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>2160p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>1440p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>1080p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>720p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>480p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>360p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>240p</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>144p</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="qualityFromPlayer">
               <property name="toolTip">
                <string>There is no guarantee this setting will be implemented by any given plugin.</string>
               </property>
               <property name="text">
                <string>Set from player</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_12">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="filtering">
      <attribute name="title">
       <string>Filtering</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QScrollArea" name="scrollArea_3">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_4">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>476</width>
            <height>87</height>
           </rect>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="leftMargin">
            <number>10</number>
           </property>
           <property name="topMargin">
            <number>10</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>10</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="spacing">
              <number>6</number>
             </property>
             <item>
              <widget class="QCheckBox" name="filterLengthCheck">
               <property name="text">
                <string>Filter videos ≤</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="filterLength">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="maximum">
                <number>5000</number>
               </property>
               <property name="singleStep">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="filterLengthLabel">
               <property name="text">
                <string>seconds</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QPushButton" name="showFilteredTerms">
               <property name="text">
                <string>Filtered Terms</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_4">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="about">
      <attribute name="title">
       <string>About</string>
      </attribute>
      <widget class="QLabel" name="qttubeLogo">
       <property name="geometry">
        <rect>
         <x>55</x>
         <y>5</y>
         <width>357</width>
         <height>80</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QLabel" name="description">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>100</y>
         <width>474</width>
         <height>164</height>
        </rect>
       </property>
       <property name="text">
        <string>TextLabel</string>
       </property>
       <property name="textFormat">
        <enum>Qt::TextFormat::RichText</enum>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignHCenter|Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
       </property>
       <property name="openExternalLinks">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QPushButton" name="saveButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
